packages/react-sdk/src/RiverContext.tsx

```diff
@@ -0,0 +1,25 @@
+'use client'
+import { SignerContext, SyncAgent } from '@river-build/sdk'
+import { createContext } from 'react'
+
+type RiverContext = {
+    signerContext: SignerContext
+    syncAgent: SyncAgent
+}
+
+export const RiverContext = createContext<RiverContext | undefined>(undefined)
+
+type RiverProviderProps = {
+    signerContext: SignerContext
+    syncAgent: SyncAgent
+    children?: React.ReactNode
+}
+
+export const RiverProvider = (props: RiverProviderProps) => {
```

Comment:
it's practical having provider and context in the same package but I've experienced annoyances with this pattern in regards to Vite & HMR causing full page reloads on changes. This won't occur unless you touch files related to the context of course but just wanted to flag this since we start fresh might as well avoid this by splitting them into two files.

Reply:
1. can we call this component RiverContextProvider?
2. Theo you just mean RiverContext.ts contains the RiverContext type and `createContext`, And RiverContextProvider.ts contains the component?

Reply:
1, 2. Yes exactly

Reply:
Thanks for the feedback and the context! I think I managed to address it